lisp/ess-inf.el

```diff
@@ -705,6 +710,21 @@ process happens interactively (when possible)."
               (ess-get-process ess-current-process-name))
           (error "Process %s is not running" name))))))
 
+(defun inferior-ess--wait-for-prompt (seconds)
```

Comment:
Having two extra functions to save one extra ">" is an overkill. We will end up with  two idle mechanisms one with :busy process property and one which looks for prompts in the inferior. I don't think it's worth it.

Reply:
I know I'm desperately perfectionistic, but I think it's worth it. The double `> >` looks very sloppy to me. It's barely 12 lines of internal code ESS can survive it ;).

Reply:
Well, I think in this case you are perfectionist about very unimportant things. You have loads of "> >  >" generated during the evaluation, why bother about extra >? How about having a perfectly clean code base instead :)  

> It's barely 12 lines of internal code ESS can survive it ;).

Well, the main thing is not the code, but the naming which apparently indicates the same intent as built-in :busy functionality.  If you really want to clean startup screen, how about the simple search-replace in the buffer? We already clean something there, it could be made more generic by fixing the double prompt plus all the potential irregularities in an uniform fashion.

Reply:
> Well, I think in this case you are perfectionist about very unimportant things. 

Of course otherwise this wouldn't be perfectionism.

I'll try the search-replace. Perhaps we could merge this as is for now with a FIXME comment as Martin wanted to release soon?

> You have loads of "> > >" generated during the evaluation

Oh those are a major bother as well, they produce unaligned outputs :(
This would be a very good reason to revisit the FIXME mentioned above.